.github/workflows/publish-package.yml

```diff
@@ -1,45 +1,44 @@
-name: Publish package to npm
+name: Publish NPM Package
 
 on:
-  workflow_dispatch:
-    branches:
-      - main
-    inputs:
-      bump:
-        description: 'bump type, major or minor or patch'
-        default: 'patch'
+  pull_request:
+    types: [closed]
 
 jobs:
-  release:
+  publish:
+    if: startsWith(github.head_ref, 'releases/') && github.event.pull_request.merged == true
```

Comment:
This is the way GitHub recommends running a workflow when a PR merges. https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#running-your-workflow-when-a-pull-request-merges